src/main/java/com/example/java/maven/cardGamePeterPan/Game.java

```diff
@@ -0,0 +1,112 @@
+                MessagePrinter messagePrinter,
+                ArrayList<Player> playerList,
+                Deck deck) {
+        this.userInputProvider = userInputProvider;
+        this.messagePrinter = messagePrinter;
+        this.playerList = playerList;
+        this.deck = deck;
+    }
+
+    public void dealCards() {
+        boolean areCards = true;
+        while (areCards) {
+            for (Player player : playerList) {
+                if (deck.getCards().size() > 0) {
+                    player.addCardToHand(deck.takeCardFromDeck());
+                }
+                if (deck.getCards().size() == 0) {
+                    areCards = false;
+                }
+            }
+        }
+    }
+
+    public void printPlayers() {
+        for (Player player : playerList) {
+            messagePrinter.printPlayer(player);
+        }
+    }
+
+    public void choseCardsToDiscard(Player player) {
+        messagePrinter.printPlayer(player);
+        messagePrinter.printMessage(ASK_FOR_CARDS_TO_DISCARD);
+        int firstCardToDiscardIndex = (getValidCardIndex(player));
+        messagePrinter.printMessage(ASK_FOR_NEXT_CARD_TO_DISCARD);
+        int secondCardToDiscardIndex = (getValidCardIndex(player));
+
+        discardCardsIfPossible(player, firstCardToDiscardIndex, secondCardToDiscardIndex);
+    }
+
+    private void discardCardsIfPossible(Player player, int firstCardToDiscardIndex, int secondCardToDiscardIndex) {
+        if (player.areCardsMatchingByRank(firstCardToDiscardIndex, secondCardToDiscardIndex)) {
+            player.discardCardsFromHand(firstCardToDiscardIndex, secondCardToDiscardIndex);
+        }
+        else if (player.areCardsMatchingBySuit(firstCardToDiscardIndex, secondCardToDiscardIndex)) {
+            player.discardCardsFromHand(firstCardToDiscardIndex, secondCardToDiscardIndex);
+        }
+        else {
+            messagePrinter.printError(WRONG_CARDS_TO_DISCARD);
+            choseCardsToDiscard(player);
+        }
+    }
+
+    public int getValidCardIndex(Player player) {
+        int userInput = userInputProvider.getNumberOfChosenCard();
+        if (userInput > 0 && userInput <= player.getHand().size()) {
+            return (userInput - 1);
+        }
+        else {
+            messagePrinter.printError(String.format("Your input must be an integer between 1 and %s", player.getHand().size()));
+            return getValidCardIndex(player);
+        }
+    }
+
+    public void getCardFromPreviousPlayerHand(Player player) {
+        Player previousPlayer = getPreviousPlayer(player);
+        messagePrinter.printMessage(String.format("%s has %s cards. Which one would you like to draw?", previousPlayer.getName(), previousPlayer.getHand().size()));
+        player.takeCardFromAnotherPlayersHand(previousPlayer, getValidCardIndex(previousPlayer));
+    }
+
+
+    public Player getPreviousPlayer(Player player) {
+        int indexOfCurrentPlayer = playerList.indexOf(player);
```

Comment:
you should get here only active players - players which have more than 0 cards. So methods can look like:
```java
private List<Player> getActivePlayers(){
        return playerList.stream()
                .filter(player -> player.getHand().size() > 0)
                .collect(Collectors.toList());
    }
```
and than you can do your magic